File: src/test/java/com/qdesrame/openapi/test/TestUtils.java
package com.qdesrame.openapi.test;

import com.qdesrame.openapi.diff.OpenApiCompare;
import com.qdesrame.openapi.diff.model.ChangedOpenApi;
import org.slf4j.Logger;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.slf4j.LoggerFactory.getLogger;

public class TestUtils {
  public static final Logger LOG = getLogger(TestUtils.class);

  public static void assertOpenApiAreEquals(String oldSpec, String newSpec) {
    ChangedOpenApi changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec);
    LOG.info("Result: {}", changedOpenApi.isChanged().getValue());
    assertTrue(changedOpenApi.getNewEndpoints().isEmpty());
    assertTrue(changedOpenApi.getMissingEndpoints().isEmpty());
    assertTrue(changedOpenApi.getChangedOperations().isEmpty());
  }

  public static void assertOpenApiChangedEndpoints(String oldSpec, String newSpec) {
    ChangedOpenApi changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec);
    LOG.info("Result: {}", changedOpenApi.isChanged().getValue());
    assertThat(changedOpenApi.getNewEndpoints()).isEmpty();
    assertThat(changedOpenApi.getMissingEndpoints()).isEmpty();
    assertThat(changedOpenApi.getChangedOperations()).isNotEmpty();
  }

  public static void assertOpenApiBackwardCompatible(
      String oldSpec, String newSpec, boolean isDiff) {
    ChangedOpenApi changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec);
    LOG.info("Result: {}", changedOpenApi.isChanged().getValue());
    assertThat(changedOpenApi.isCompatible()).isTrue();
  }

  public static void assertOpenApiBackwardIncompatible(String oldSpec, String newSpec) {
    ChangedOpenApi changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec);
    LOG.info("Result: {}", changedOpenApi.isChanged().getValue());
    assertThat(changedOpenApi.isIncompatible()).isTrue();
  }
}

File: src/test/java/com/qdesrame/openapi/test/TestUtils.kt
package com.qdesrame.openapi.test

import com.qdesrame.openapi.diff.OpenApiCompare
import org.junit.jupiter.api.Assertions
import org.slf4j.LoggerFactory

object TestUtils {
    private val LOG = LoggerFactory.getLogger(TestUtils::class.java)
    
    @JvmStatic
    fun assertOpenApiAreEquals(oldSpec: String, newSpec: String) {
        val changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec)
        LOG.info("Result: {}", changedOpenApi.isChanged.value)
        Assertions.assertTrue(changedOpenApi.newEndpoints.isEmpty())
        Assertions.assertTrue(changedOpenApi.missingEndpoints.isEmpty())
        Assertions.assertTrue(changedOpenApi.changedOperations.isEmpty())
    }

    @JvmStatic
    fun assertOpenApiChangedEndpoints(oldSpec: String, newSpec: String) {
        val changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec)
        LOG.info("Result: {}", changedOpenApi.isChanged.value)
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.newEndpoints).isEmpty()
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.missingEndpoints).isEmpty()
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.changedOperations).isNotEmpty
    }

    @JvmStatic
    fun assertOpenApiBackwardCompatible(oldSpec: String, newSpec: String, isDiff: Boolean) {
        val changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec)
        LOG.info("Result: {}", changedOpenApi.isChanged.value)
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.isCompatible).isTrue()
    }

    fun assertOpenApiBackwardCompatibleFromContent(oldSpec: String, newSpec: String) {
        val changedOpenApi = OpenApiCompare.fromContents(oldSpec, newSpec)
        LOG.info("Result: {}", changedOpenApi.isChanged.value)
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.isCompatible).isTrue()
    }

    @JvmStatic
    fun assertOpenApiBackwardIncompatible(oldSpec: String, newSpec: String) {
        val changedOpenApi = OpenApiCompare.fromLocations(oldSpec, newSpec)
        LOG.info("Result: {}", changedOpenApi.isChanged.value)
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.isIncompatible).isTrue()
    }

    fun assertOpenApiBackwardIncompatibleFromContent(oldSpec: String, newSpec: String) {
        val changedOpenApi = OpenApiCompare.fromContents(oldSpec, newSpec)
        LOG.info("Result: {}", changedOpenApi.isChanged.value)
        org.assertj.core.api.Assertions.assertThat(changedOpenApi.isIncompatible).isTrue()
    }
}